File: 0053-maximum-subarray/0053-maximum-subarray.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var maxSubArray = function(nums) {
    let maxSum=nums[0];
    let currSum=0;

    for(let num of nums){
        currSum+=num;
        maxSum=Math.max(currSum, maxSum);

        if(currSum < 0){
            currSum=0;
        }
    }
    return maxSum;

};

File: 0053-maximum-subarray/0053-maximum-subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
       int maxSum=nums[0];
       int currSum=0;

       for(int num:nums){
          currSum+=num;
          maxSum=max(maxSum, currSum);
          if(currSum < 0){
            currSum=0;
          }

       } 
       return maxSum;
    }
};